src/java/org/apache/cassandra/db/compaction/TimeWindowCompactionStrategy.java

```diff
@@ -56,6 +58,7 @@ public class TimeWindowCompactionStrategy extends AbstractCompactionStrategy
     private final Set<SSTableReader> sstables = new HashSet<>();
     private long lastExpiredCheck;
     private long highestWindowSeen;
+    private Map<Long, Integer> sstableCountByBuckets = Collections.emptyMap();
```

Comment:
if `sstableCountByBuckets` is accessed directly by a JMX thread, I think we want to make it volatile here.

Reply:
Subtle; good catch. This definitely opens this up to being accessed by multiple thread contexts. This tweak should give us slightly more up-to-date information in what we see via JMX if it just so happens to race w/the repair manager or compaction managers picking their next tasks. 👍